src/views/TheCodeUploader.vue

```diff
@@ -0,0 +1,106 @@
+<template lang='pug'>
+	v-container(class='fill-height', fluid)
+		v-row(
+			ref='dropArea',
+			:class="{ 'dragged-over': isDraggedOver }",
+			class='drop-area',
+			justify='center',
+			align='center',
+			@dragenter='isDraggedOver = true',
+			@dragleave='isDraggedOver = $refs.dropArea.contains($event.relatedTarget)',
+			@drop.prevent='handleFilesUpload',
+			@dragover.prevent,
+			)
+			div(class='d-flex flex-column ma-4')
+				h3(class='headline font-weight-regular text-center') Upload files using field below or by dragging it on this area.
+				v-form(ref='form', class='mt-2')
+					v-file-input(
+						v-model='files',
+						class='mx-6',
+						accept='.js',
+						autofocus,
+						backgroundColor='white',
+						chips,
+						color='green',
+						multiple,
+						outlined,
+						placeholder='Click here to upload file...',
+						:rules='fileInputRules',
+						:show-size='1024',
+						@change='handleFilesUpload',
+						)
+				v-btn(class='px-10 align-self-center', color='green', dark, large, :loading='isLoading', @click='detectSmells') Detect code smells
+
+		v-row(v-show='detectorResult.length', ref='smellsContainer', class='pt-12' justify='center', no-gutters)
+			v-col(class='d-flex flex-column', md='12', lg='11', xl='10')
+				h3(class='headline font-weight-regular mb-1') Detected code smells in files
+				base-files-smells-expansion-panels(:detectorResult='detectorResult')
+
+		base-snackbar(ref='snackbar')
+</template>
+
+<script>
+import { api } from '../axios';
+import { uniqWith } from 'lodash';
+
+export default {
+	name: 'TheCodeUploader',
+
+	data: () => ({
+		files: [],
+		fileInputRules: [
+			value => !!value.length || 'In order to run detection, it\'s required to provide some files.',
+			value => value.every(file => file.type === 'text/javascript') || 'Only JavaScript files are allowed.',

```

Comment:
I would not rely on MIME type as it heavily depends on the OS configuration. I've tested it under Ubuntu Linux and it doesn't work. Instead, let's just validate the file extension:
```javascript
value => value.every(file => file.endsWith('.js')) || 'Only JavaScript files are allowed.',
```

Reply:
Thanks for resolving these problems. :)